### ISSUE TITLE
[Question] show own answers for logged-in fe-users

### ISSUE THREAD

**Issue by medarob** [bihor/fp_masterquiz#40]
Hi,

I wanted to ask if it is already possible to show the answers for a particular user and quiz on a separate page.

In our example a fe-user answers the questions. After answering all questions, the user can't change anything and also don't sees the choosen options. So it would be great to have the possibility to show the answers from a logged-in fe-user.

Maybe on the quiz page.
Instead of just showing the text "You already have participated for this quiz" the already given answers could be shown below the text as well? But maybe that is already possible?

**Comment by bihor** [comment]
You can define many things. It is possible to show the result of the current user at the end of the quiz. See pics and manual.
![quiz1](https://user-images.githubusercontent.com/8191408/186629075-295d27a6-fb32-4144-afa2-fed15264b843.jpg)
![quiz2](https://user-images.githubusercontent.com/8191408/186629081-9d8760dc-991c-4202-be48-c6d8c430450d.jpg)

https://docs.typo3.org/p/fixpunkt/fp-masterquiz/3.4/en-us/User/Index.html#actions

Do you want to show results of other users?

**Comment by bihor** [comment]
PS: you find a example here:
https://www.quizpalme.de/quiz-umfrage/wissens-quizze/romantia-quiz

**Comment by medarob** [comment]
No, I just want to show the results from the current user when this user tries to do the quiz again, so not from other users.

It seems that the condition `<f:if condition="{showAnswers}">` was changed in our own templates and the whole code in the `<f:if>` clause was set in comments and replaced by the text `"Sie haben bereits teilgenommen."`. (I thought that comes from the extension.)
If I undo the comment around the code then the answers are shown from the current user.

I also noticed/wondered that the option `showOwnAnswers` seems to have no impact here. Either 0 or 1 changes nothing and the answers are shown when I try to do the quiz again.
In our extension the following settings were overridden with 0 but I don't know why:
```
        showAnswerPage = 0
        showOwnAnswers = 0
        showCorrectAnswers = 0
        showEveryAnswer = 0
        showAnswersAtFinalPage = 0
        showAllAnswers = 0
```

If I change `showOwnAnswers` to 1 then there's also no change in the FE. Answers are still be shown. (Which is the final result I want but shouldn't 0 disable the answers and 1 display the answers?)

The docu says for this option "Show the answers of the quiz taker?" (Which is what I want) but it doesn't say where it will be shown? Is it for a different page or is it the same like the quiz page?

**Comment by bihor** [comment]
> Note: if you enable a cookie or FEuser check, then an user cannot vote again, if he had already voted/participated. The participant will see his result of a poll/quiz instead of the checkboxes/radio buttons.

Maybe this is the reason why showOwnAnswers  has no effect.

**Comment by medarob** [comment]
Yes, the FEuser check is enabled.

So, the option `showOwnAnswers` has only an effect when the FEuser check is disabled?

**Comment by bihor** [comment]
I think that is correct for a answered quiz.
It should work for a not yet answered quiz.
